TradeDis/frontend
# feat:update post

**What was done?**
Story: TRAD-98 "As a user, I would like to mark my posts as inactive once a trade has been done."

- Created a switch UI element for the original poster (and only the original poster) to be able to mark their post as "active" or "inactive"
- Update post's status in database 
- Update home feed to distinguish inactive posts from active posts

**Areas which need special reviewing**
- Only the original poster is able to view the switch on the post and for that I used the `AuthContext` provider. I haven't used this before so I'm not sure if this is the best way to do it.
- Inactive posts on the home feed show up darkened and with text strikethrough, we might want to style inactive elements differently in the future.

**Testing**
This test ensures that the status is being correctly updated in the database and that the styles on the home feed are being applied to inactive posts.

1. Ensure the backend server is running (`yarn dev` in the backend project)
2. Run the frontend on a simulator (`expo start`)
3. Log into the application with an existing user
4. Navigate to the create post page and create a post (if you don't already have a post created)
5. Search for the created post on the home feed
6. Navigate to the post's details
7. You should see the switch element on the top right of the screen
8. Switch the post's status to "inactive"
9. Request the endpoint `/api/v1/posts` and confirm that the status of your post has been updated
10. Return back to the home feed and confirm that the styling of the inactive post has changed (darker background and strikethrough text)
11. Navigate back to the post's details page and switch the status back to "active"
12. Request the endpoint `/api/v1/posts` and confirm that the status of your post has been updated
13. Return back to the home feed and confirm that the styling of the post has gone back to normal
